programs/clearing_house/src/controller/orders.rs

```diff
@@ -1079,12 +1079,24 @@ fn fulfill_order(
         return Ok((0, false, true));
     }
 
-    let fulfillment_methods = determine_perp_fulfillment_methods(
-        &user.orders[user_order_index],
-        maker.is_some(),
-        amm_is_available,
-        slot,
-    )?;
+    let fulfillment_methods = {
+        let market = perp_market_map.get_ref(&market_index)?;
+
+        determine_perp_fulfillment_methods(
+            &user.orders[user_order_index],
+            if maker.is_some() {
```

Comment:
```
            if let Some(maker) = maker {
                Some(&maker.orders[maker_order_index.unwrap()])
            } else {
                None
            },
```

a bit nicer to read imo and `iter` usually means iterator in rust, so a bit weird